src/jrd/tra.cpp

```diff
@@ -143,7 +143,13 @@ void TRA_attach_request(Jrd::jrd_tra* transaction, Jrd::jrd_req* request)
 void TRA_detach_request(Jrd::jrd_req* request)
 {
 	if (!request->req_transaction)
+	{
+		fb_assert(!request->req_savepoints);
 		return;
+	}
+
+	// Release stored looper savepoints
+	Savepoint::destroy(request->req_savepoints);
```

Comment:
If you wish, here it is:
`	ntdll.dll!000007f8f620783c()	Unknown`
`	engine13.dll!Firebird::Mutex::enter(const char * aReason) Line 111	C++`
` 	engine13.dll!Firebird::EnsureUnlock<Firebird::Mutex,Firebird::NotRefCounted<Firebird::Mutex> >::enter() Line 130	C++`
` 	engine13.dll!Firebird::MemPool::releaseBlock(Firebird::MemBlock * block) Line 2150	C++`
` 	engine13.dll!Firebird::MemPool::release(void * object, bool flagDecr) Line 2125	C++`
` 	engine13.dll!Firebird::MemPool::deallocate(void * block) Line 2361	C++`
` 	engine13.dll!Firebird::MemPool::globalFree(void * block) Line 2349	C++`
` 	engine13.dll!Firebird::MemoryPool::globalFree(void * block) Line 2490	C++`
` 	engine13.dll!operator delete(void * mem) Line 355	C++`
` 	[External Code]	`
` 	engine13.dll!Jrd::Savepoint::destroy(Jrd::Savepoint * & savepoint) Line 229	C++`
` 	engine13.dll!TRA_detach_request(Jrd::jrd_req * request) Line 155	C++`
` 	engine13.dll!EXE_unwind(Jrd::thread_db * tdbb, Jrd::jrd_req * request) Line 977	C++`
` 	engine13.dll!EXE_release(Jrd::thread_db * tdbb, Jrd::jrd_req * request) Line 717	C++`
` 	engine13.dll!Jrd::JrdStatement::release(Jrd::thread_db * tdbb) Line 577	C++`
` 	engine13.dll!CMP_release(Jrd::thread_db * tdbb, Jrd::jrd_req * request) Line 448	C++`
` 	engine13.dll!Jrd::dsql_req::destroy(Jrd::thread_db * tdbb, Jrd::dsql_req * request, bool drop) Line 1610	C++`
` 	engine13.dll!DSQL_free_statement(Jrd::thread_db * tdbb, Jrd::dsql_req * request, unsigned short option) Line 321	C++`
` 	engine13.dll!Jrd::JStatement::freeEngineData(Firebird::CheckStatusWrapper * user_status) Line 4861	C++`
` 	engine13.dll!Jrd::JStatement::free(Firebird::CheckStatusWrapper * user_status) Line 4882	C++`
` engine13.dll!Firebird::IStatementBaseImpl<Jrd::JStatement,Firebird::CheckStatusWrapper,Firebird::IReferenceCountedImpl<Jrd::JStatement,Firebird::CheckStatusWrapper,Firebird::Inherit<Firebird::IVersionedImpl<Jrd::JStatement,Firebird::CheckStatusWrapper,Firebird::Inherit<Firebird::IStatement> > > > >::cloopfreeDispatcher(Firebird::IStatement * self, Firebird::IStatus * status) Line 8242	C++`
` 	fbclient.dll!Firebird::IStatement::free<Firebird::CheckStatusWrapper>(Firebird::CheckStatusWrapper * status) Line 1661	C++`
 `	fbclient.dll!Why::YStatement::free(Firebird::CheckStatusWrapper * status) Line 4423	C++`
` 	fbclient.dll!Firebird::IStatementBaseImpl<Why::YStatement,Firebird::CheckStatusWrapper,Firebird::IReferenceCountedImpl<Why::YStatement,Firebird::CheckStatusWrapper,Firebird::Inherit<Firebird::IVersionedImpl<Why::YStatement,Firebird::CheckStatusWrapper,Firebird::Inherit<Firebird::IStatement> > > > >::cloopfreeDispatcher(Firebird::IStatement * self, Firebird::IStatus * status) Line 8242	C++`
 `	firebird.exe!Firebird::IStatement::free<Firebird::CheckStatusWrapper>(Firebird::CheckStatusWrapper * status) Line 1661	C++`
` 	firebird.exe!rem_port::end_statement(p_sqlfree * free_stmt, packet * sendL) Line 3000	C++`
 `	firebird.exe!process_packet(rem_port * port, packet * sendL, packet * receive, rem_port * * result) Line 4389	C++`
` 	firebird.exe!loopThread(void * __formal) Line 5817	C++`
 `	firebird.exe!`anonymous namespace'::ThreadArgs::run() Line 78	C++`
 `	firebird.exe!threadStart(void * arg) Line 94	C++`

Actual reason of AV is unintialized spinLock:
`+		DebugInfo	0x0000000000000000 <NULL>	_RTL_CRITICAL_SECTION_DEBUG *`
`		LockCount	0x00000000	long`
`		RecursionCount	0x00000000	long`
`		OwningThread	0x0000000000000000	void *`
`		LockSemaphore	0xcccccccccccccccc	void *`
`		SpinCount	0xcccccccccccccccc	unsigned __int64`